kythe/java/com/google/devtools/kythe/platform/java/JavaCompilationDetails.java

```diff
@@ -179,8 +180,8 @@ public JavacTask getJavac() {
   }
 
   /** Returns the AST for the current analysis target. */
-  public Iterable<? extends CompilationUnitTree> getAsts() {
-    return asts;
+  public Optional<Iterable<? extends CompilationUnitTree>> getAsts() {
```

Comment:
Do we really need to distinguish "not there" from "empty iterable"? Why not

  return asts == null ? ImmutableList.of() : asts;

Reply:
👍